src/MCPClient/lib/clientScripts/archivematicaClamscan.py

```diff
@@ -115,19 +115,38 @@ def scan(self, path):
         try:
             result = getattr(self, method_name)(path)
             state, details = result[result_key]
-        except IOError as err:
-            if err.errno == errno.EPIPE:
-                logger.error(
-                    '[Errno 32] Broken pipe. File not scanned. Check Clamd '
-                    'StreamMaxLength')
-                return None, state, details
         except Exception as err:
-            logger.error('Virus scanning failed: %s', err, exc_info=True)
+            return self.clamd_exception_handler(Exception, err), state, details
```

Comment:
There's no need to pass in the `Exception` class here. It's just a globally available var anyways and you're not using it. Also, I think you should make this a function or a `@staticmethod` and remove the `self` since it's not being used.

Also, in general catching `Exception` is frowned upon in Python. Why not do `except (IOError, BufferTooLongError, ConnectionError) as err:` instead, assign to `passed` in the scope of that `except`, and then put the sole `return` call of `scan` in a `finally:` block.

Reply:
Good catch, I kept saying to myself to remove that! - Have added a finally statement. Hopefully looking a lot better. Much appreciated.